src/shared/components/palette/case-flag/components/update-flag/update-flag.component.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { CaseFlagFieldState, UpdateFlagErrorMessage } from '../../enums';
+
+import { UpdateFlagComponent } from './update-flag.component';
+
+describe('UpdateFlagComponent', () => {
+  let component: UpdateFlagComponent;
+  let fixture: ComponentFixture<UpdateFlagComponent>;
+  let nextButton: any;
+  let textareaInput: string;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ ReactiveFormsModule ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ],
+      declarations: [ UpdateFlagComponent ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UpdateFlagComponent);
+    component = fixture.componentInstance;
+    component.formGroup = new FormGroup({});
+    nextButton = fixture.debugElement.nativeElement.querySelector('button[type="button"]');
+    // 200-character text input
+    textareaInput = '0000000000' + '1111111111' + '2222222222' + '3333333333' + '4444444444' + '5555555555' + '6666666666' +
```

Comment:
@tarikdogan-hmcts - this is only 100 characters long... you need to extend it

Reply:
done